### PR TITLE
Improve defaults and docs

### LINKED ISSUE #410: Set new defaults
For the new MALA version we should decide what options we want to have as a default. E.g.: Should OpenPMD be the default value, should certain performance enhancements (see #409) be made the default, etc.; MALA should keep a balance between performance and stabilitiy.

### PR DESCRIPTION
**Pull request by RandomDefaultUser**
This PR massively overhauls the documentation and examples, moving towards MALA v1.2.0. 
Closes #410 .

### CODE REVIEW

**Comment by acangi** [comment]
Remarks on the updated documentation:
# Landing page
- Add "How to cite MALA" 
  - Also we need to decide which papers (or DOIs) we would like user of MALA to cite
- Rename "Where is MALA"
  - Instead we could list papers where MALA has been used (along the lines of what's there already) and add a short description about each of these papers.
- Mention HZDR together with CASUS
- A few more minor language style changes on the landing page

**Comment by elcorto** [comment]
* [ ] python -> Python everywhere
* [ ] remove "simply" everywhere :)
* [ ] `install/installing_mala.html`
  * [ ] "This the installation of to ..."
  * [ ] is there a pyproject.toml now?
    * [ ] `pip install -e .[doc]` vs. `pip install -r docs/requirements.txt` ?
  * [ ] `test-data` latest tag is 1.7.0, docs say 1.6.0
* [ ] `install/installing_lammps.html`
  * [ ] "further options you will have to set further options:.."
  * [ ] `python3 install.py -p lammps -l ../<build_folder>/liblammps.so -v
    ../src/version.h`: what about `-w $(pwd)` as in
    https://github.com/mala-project/mala/issues/456 and
    https://github.com/mala-project/mala/issues/455
  * [ ] how about one example copy-paste command line for GPU and one for CPU (like
    https://github.com/mala-project/mala/issues/455#issuecomment-1561868121)
    for CPU+OpenMP
  * [ ] do we need the same C compiler that was also used to build Python?, in case
    of MPI, should the MPI compiler wrapper use the required C compiler?
* [ ] `install/installing_qe.html`
  * [ ] "the most recent version upon this release of"
* [ ] `basic_usage/trainingmodel.html`
  * [ ] "with in the following."
  * [ ] why "test_" in "test_network", "test_trainer"?
* [ ] `basic_usage/more_data.html`
  * [ ] "Fine-granular access to the calculators is enabled via the descriptor_calculation_kwargs and target_calculation_kwargs arguments, but usually not needed."    use `descriptor_calculation_kwargs` and `target_calculation_kwargs`
* [ ] `basic_usage/predictions.html`
  * [ ] "you have to provide to a pseudopotential"
* [ ] `advanced_usage/trainingmodel.html`
  * [ ] "will* adversely affect .." - only one `*`

**Comment by RandomDefaultUser** [comment]
- To-Dos from meeting:
	- [ ] Write short descriptions of all papers
	- [ ] Add a "Citing MALA" tab

**Comment by RandomDefaultUser** [comment]
- [ ] Add Franz, Tim, Bartek, Hossein to contributors

File: docs/source/advanced_usage/openpmd.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            ...
+            # Changes for DataShuffler
+            data_shuffler = mala.DataShuffler(parameters)
+            # Data can be shuffle FROM and TO openPMD - but also from
+            # numpy to openPMD.
+            data_shuffler.add_snapshot("Be_snapshot0.in.h5", data_path,
+                                        "Be_snapshot0.out.h5", data_path,
+                                        snapshot_type="openpmd")
+            data_shuffler.shuffle_snapshots(...,
+                                            save_name="Be_shuffled*.h5")
```
Comment (franzpoeschel) [inline docs/source/advanced_usage/openpmd.rst:47]:
```suggestion
                                            save_name="Be_shuffled*.h5")


For further information on the interaction with openPMD data,
please consult the `official documentation <https://openpmd-api.readthedocs.io>`_.

As a user of Mala, you will be mainly interested in the scientific tooling that can read openPMD, e.g.:

* visualization and analysis, including an exploratory Jupyter notebook GUI: `openPMD-viewer <https://github.com/openPMD/openPMD-viewer>`_
* `yt-project <http://yt-project.org/doc/examining/loading_data.html#openpmd-data>`_ (`tutorial <https://gist.github.com/C0nsultant/5808d5f61b271b8f969d5c09f5ca91dc>`_)
* `ParaView <https://www.paraview.org/>`_ has a Python-based openPMD plugin that can be activated by opening a helper text file ending on ``.pmd`` that contains one line with the openPMD-api Series filename, e.g. ``data_%T.bp``
* `VisIt <https://github.com/openPMD/openPMD-visit-plugin>`_
* converter tools: `openPMD-converter <https://github.com/openPMD/openPMD-converter>`_
* full list of `projects using openPMD <https://github.com/openPMD/openPMD-projects>`_

If you intend to write your own post-processing routines, make sure to check out our `example files <https://github.com/openPMD/openPMD-example-datasets>`_ and the `formal, open standard <https://github.com/openPMD/openPMD-standard>`_ on openPMD.
```

**Review by franzpoeschel — COMMENTED** [review]
Looks good to me. I've fixed some inconsistencies and typos, and also added a section with further links for documentation and tooling.

I will be on vacation next week, so feel free to edit and use as you please.

File: docs/source/advanced_usage/openpmd.rst (hunk 1 above)
Comment (franzpoeschel) [reply docs/source/advanced_usage/openpmd.rst:47]: (This is based on a slightly updated version of https://picongpu.readthedocs.io/en/latest/postprocessing/openPMD.html)

File: docs/source/CONTRIBUTE.md [hunk 2]
```diff
@@ -1,9 +1,35 @@
-# Contributing to MALA
-
-MALA is open source software and as such always welcomes additions and
-improvements. However, we kindly ask any contributor to adhere to the following
-suggestions in order to keep the overall quality and maintainability of the
-code high.
+# Contributions
+
+MALA is open source software and based on the work of many people.
```
Comment (acangi) [inline docs/source/CONTRIBUTE.md:3]: MALA is an open-source software and is built upon the collaborative efforts of many contributors. The MALA team warmly welcomes additional contributions and kindly requests potential contributors to follow the suggested guidelines below to ensure the code's overall quality and maintainability.

File: docs/source/CONTRIBUTE.md [hunk 3]
```diff
@@ -1,9 +1,35 @@
...
+# Contributions
+
+MALA is open source software and based on the work of many people. 
+The MALA team always welcomes contributions, but kindly asks potential 
+contributors to adhere to the following suggestions in order to keep the 
+overall quality and maintainability of the code high.
+
+## MALA contributors
+
+Many people have contributed to MALA, and we are deeply thankful for it.
```
Comment (acangi) [inline docs/source/CONTRIBUTE.md:10]: Many people have made valuable contributions to MALA, and we are immensely grateful for their support.

File: docs/source/index.md [hunk 4]
```diff
@@ -1,21 +1,99 @@
-# Welcome to MALA's documentation!
+![image](./img/logos/mala_horizontal.png)
+
+# Welcome to MALA!
+
+## What is MALA?
+
+MALA is a software package for building ML models that replace
+density functional theory (DFT) calculations. DFT is **the** most
+popular method for simulating materials on a quantum level and predicting
```
Comment (acangi) [inline docs/source/index.md:9]: DFT is one of the most widely used methods for simulating materials at a quantum level and predicting their properties, employed by researchers worldwide.

File: docs/source/index.md [hunk 5]
```diff
@@ -1,21 +1,99 @@
...
+MALA is a software package for building ML models that replace
+density functional theory (DFT) calculations. DFT is **the** most
+popular method for simulating materials on a quantum level and predicting 
+materials properties, and employed by researcher throughout the world. 
+
+While DFT is very efficient compared to other quantum mechanical simulation
+techniques, it still scales cubically with number of atoms and temperature 
+simulated, making large scale simulation at arbitrary temperatures very 
+computationally challenging. 
+
```
Comment (acangi) [inline docs/source/index.md:16]: Although DFT is more efficient compared to other quantum mechanical simulation techniques, it still exhibits cubic scaling with the number of atoms. This limitation poses significant computational challenges for conducting large-scale simulations.

File: docs/source/index.md [hunk 6]
```diff
@@ -1,21 +1,99 @@
...
+Therefore, a new direction of research tries to [combine the power of ML with DFT](https://journals.aps.org/prmaterials/abstract/10.1103/PhysRevMaterials.6.040301).
+MALA is at the forefront of these efforts, being the first library to allow 
+users to build ML-DFT models that give **full access to the electronic structure**,
+observables such as the total free energy of the system **and** scale well with
+system size and temperature.
+
```
Comment (acangi) [inline docs/source/index.md:22]: Recent research activities focus on harnessing the [potential of ML and DFT](https://journals.aps.org/prmaterials/abstract/10.1103/PhysRevMaterials.6.040301). MALA leads these efforts by facilitating the construction of ML-DFT models, offering comprehensive access to the electronic structure and observables, including the total free energy of the system. Moreover, MALA ensures scalability with system size and various parameters, such as temperature.

File: docs/source/index.md [hunk 7]
```diff
@@ -1,21 +1,99 @@
...
+## How does MALA work? 
+
+MALA operates by predicting the so called local density of states (LDOS). 
+The LDOS is an efficient and versatile representation of the electronic 
+structure of a material. Based on the LDOS, a range of useful observables can
+be calculated. 
+
+MALA calculates a grid-based representation of any atomic structure presented
```
Comment (acangi) [inline docs/source/index.md:30]: MALA calculates a grid-based representation for any given atomic structure.

File: docs/source/index.md [hunk 8]
```diff
@@ -1,21 +1,99 @@
...
+## How does MALA work? 
+
+MALA operates by predicting the so called local density of states (LDOS). 
+The LDOS is an efficient and versatile representation of the electronic 
+structure of a material. Based on the LDOS, a range of useful observables can
+be calculated. 
+
+MALA calculates a grid-based representation of any atomic structure presented 
+to it (so called descriptors). These descriptors are then passed through a 
+neural network, a very efficient ML algorithm, to predict the LDOS. This is
```
Comment (acangi) [inline docs/source/index.md:32]: These descriptors are subsequently fed into a neural network, an efficient ML algorithm, to predict the LDOS.

File: docs/source/index.md [hunk 9]
```diff
@@ -1,21 +1,99 @@
...
+The LDOS is an efficient and versatile representation of the electronic 
+structure of a material. Based on the LDOS, a range of useful observables can
+be calculated. 
+
+MALA calculates a grid-based representation of any atomic structure presented 
+to it (so called descriptors). These descriptors are then passed through a 
+neural network, a very efficient ML algorithm, to predict the LDOS. This is 
+done individually at each point of the real space grid, allowing for easy 
+scalability and parallelization. 
+
```
Comment (acangi) [inline docs/source/index.md:35]: The prediction is performed individually at each point of the real space grid, facilitating straightforward scalability and parallelization.

File: docs/source/index.md [hunk 10]
```diff
@@ -1,21 +1,99 @@
...
+Afterwards, a range of post-processing features are available within MALA to
+calculate obervables from interest. Just like a DFT calculation - just faster.
+For more detail on the workflow, check out the [first MALA publication](https://www.doi.org/10.1103/PhysRevB.104.035120).
+
```
Comment (acangi) [inline docs/source/index.md:39]: Subsequently, MALA offers a variety of post-processing features to compute desired observables. Similar to a DFT calculation, but with enhanced speed. For further details on the workflow, refer to the [first MALA publication](https://www.doi.org/10.1103/PhysRevB.104.035120).

File: docs/source/index.md [hunk 11]
```diff
@@ -1,21 +1,99 @@
...
+(Scientific supervisor: Sivasankaran Rajamanickam, code maintenance: 
+Jon Vogel)
+and the [Center for Advanced Systems Understanding](https://www.casus.science/), 
+an institute of the [Helmholtz-Zentrum Dresden-Rossendorf](https://www.hzdr.de/) 
+(Scientific supervisor: Attila Cangi, code maintenance: Lenz Fiedler).
+
+## MALA publications 
+
+MALA has been used in a range of publications:
+
```
Comment (acangi) [inline docs/source/index.md:62]: MALA has been employed in various publications, showcasing its versatility and effectiveness.

File: README.md
Comment (acangi) [inline README.md:1]: In line 13, I would add "an institute of the Helmholtz-Zentrum Dresden-Rossendorf" after ... (CASUS).

**Pull request merged**